packages/core/src/utils/check-boundaries.ts

```diff
@@ -0,0 +1,58 @@
+import { isNumber } from './types';
+
+export interface NumBoundaries {
+  min?: number;
+  max?: number;
+  exclusiveMin?: boolean;
+  exclusiveMax?: boolean;
+}
+
+const checkMinimum = (
+  value: number,
+  options: Omit<NumBoundaries, 'max' | 'exclusiveMax'> = {}
+): boolean => {
+  const exclusiveMin = !!options.exclusiveMin;
+  const min = options.min ?? Number.MIN_SAFE_INTEGER;
+
+  return exclusiveMin ? value > min : value >= min;
+};
+
+const checkMaximum = (
+  value: number,
+  options: Omit<NumBoundaries, 'min' | 'exclusiveMin'> = {}
+): boolean => {
+  const exclusiveMax = !!options.exclusiveMax;
+  const max = options.max ?? Number.MAX_SAFE_INTEGER;
+
+  return exclusiveMax ? value < max : value <= max;
+};
+
+export const checkBoundaries = (
+  value: unknown,
+  options: {
+    min?: number;
+    max?: number;
+    exclusiveMin?: boolean;
+    exclusiveMax?: boolean;
+  } = {}
+) => {
+  if (typeof value === 'string') {
+    value = parseInt(value, 10);
+  }
+
+  if (!isNumber(value)) {
```

Comment:
```suggestion
  if (!Number.isSafeInteger(value)) {
```
(if checkBoundaries(Number.MAX_VALUE) is expected to return false)

Reply:
The question is should we accept floats as well? If so, we can't utilize this check

Reply:
from float-support pov - now references to `Number.MIN_SAFE_INTEGER` & `Number.MAX_SAFE_INTEGER` in `checkMinimum()` & `checkMaximum()` look invalid (though there are never take place as `checkMinimum`/`checkMaximum` is always called with defined `min`/`max` atm).
imo it's better to return `true` from that `checkX` if `X` boundary is missing instead of using artificial boundaries.